# Identitas Penulis
- **Nama Lengkap**: Muhammad Ikctiar Saputra
- **NIM**: 250401020169
- **Kelas**: IF401
- **Program Studi**: PJJ Informatika

In [1]:
import pandas as pd
import numpy as np

URL = "https://raw.githubusercontent.com/mwaskom/seaborn-data/master/tips.csv"
# Output dari pd.read_csv adalah DataFrame
df = pd.read_csv(URL)
print("Dataset berhasil dimuat.")
print("Shape dataset:", df.shape)

Dataset berhasil dimuat.
Shape dataset: (244, 7)


In [2]:
# Berapa jumlah baris dan kolom dataset Tips?
# Dapat dijawab menggunakan DataFrame.shape, output-nya (jumlah_baris, jumlah_kolom)
df_shape = df.shape
jumlah_baris = df_shape[0]
jumlah_kolom = df_shape[1]
print('Jumlah baris:', jumlah_baris)
print('Jumlah kolom:', jumlah_kolom)

Jumlah baris: 244
Jumlah kolom: 7


In [3]:
# Kolom mana yang memiliki nilai kosong (missing)?
# Dapat dijawab menggunakan DataFrame.info()
print(df.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 244 entries, 0 to 243
Data columns (total 7 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   total_bill  244 non-null    float64
 1   tip         244 non-null    float64
 2   sex         244 non-null    object 
 3   smoker      244 non-null    object 
 4   day         244 non-null    object 
 5   time        244 non-null    object 
 6   size        244 non-null    int64  
dtypes: float64(2), int64(1), object(4)
memory usage: 13.5+ KB
None


In [4]:
# Berapa persen pelanggan yang merokok (smoker = 'Yes')?
# Tahapannya:
# 1. Filter baris dengan kondisi smoker == 'Yes'
# 2. Hitung rasio terhadap total baris dan kalikan 100
smokers = df[df['smoker'] == 'Yes'].shape[0]
total = df.shape[0]
persen_smoker = (smokers / total) * 100
print(f"Jumlah pelanggan merokok: {smokers} dari {total}")
print(f"Persentase pelanggan merokok: {persen_smoker:.2f}%")

Jumlah pelanggan merokok: 93 dari 244
Persentase pelanggan merokok: 38.11%


In [5]:
# Rata-rata tip yang diberikan oleh pelanggan wanita (sex = 'Female') pada waktu makan malam (time = 'Dinner')?
# Tahapannya:
# 1. Buat DataFrame terfilter dengan kondisi sex == 'Female' dan time == 'Dinner'
# 2. Ambil kolom 'tip' dan hitung rata-rata menggunakan fungsi .mean()
df_filtered = df[(df['sex'] == 'Female') & (df['time'] == 'Dinner')]
avg_tip_female_dinner = df_filtered['tip'].mean()
print(f"Jumlah transaksi wanita saat dinner: {df_filtered.shape[0]}")
print(f"Rata-rata tip: ${avg_tip_female_dinner:.2f}")

Jumlah transaksi wanita saat dinner: 52
Rata-rata tip: $3.00


In [6]:
# Hari apa yang rata-rata total tagihannya (total_bill) paling tinggi?
# Tahapannya:
# 1. Lakukan grouping berdasarkan kolom 'day' menggunakan groupby()
# 2. Ambil kolom 'total_bill' dan panggil fungsi mean()
avg_bill_per_day = df.groupby('day')['total_bill'].mean().sort_values(ascending=False)
print("Rata-rata total bill per hari:")
print(avg_bill_per_day)

Rata-rata total bill per hari:
day
Sun     21.410000
Sat     20.441379
Thur    17.682742
Fri     17.151579
Name: total_bill, dtype: float64


## Kesimpulan
Pada pertemuan kedua ini, saya mempelajari dasar-dasar Eksplorasi Data Analitis (EDA) dengan menggunakan pustaka Pandas:
1. Membaca data tabular langsung dari link eksternal menggunakan `pd.read_csv()`.
2. Memahami dimensi dataset menggunakan properti `.shape`.
3. Memeriksa ringkasan tipe data dan mendeteksi adanya data kosong (missing values) menggunakan fungsi `.info()` (pada dataset Tips ini tidak ditemukan nilai kosong).
4. Melakukan penyaringan baris (*row filtering*) berdasarkan kondisi kolom kategorikal tunggal maupun gabungan (menggunakan operator logika `&`).
5. Menggunakan agregasi data kelompok dengan `.groupby()` untuk menganalisis perbandingan nilai rata-rata antar kategori.

**Temuan Utama**: Rata-rata total tagihan tertinggi di restoran terjadi pada hari Minggu ($21.41), diikuti hari Sabtu ($20.44). Pelanggan perokok mencakup 38.11% dari keseluruhan entri data.

**Keterbatasan/Pertanyaan**: Bagaimana jika dataset memiliki data yang kotor seperti data duplikat, outlier, atau nilai kosong yang signifikan? Langkah-langkah apa yang harus diambil untuk merapikannya?